holoviews/plotting/renderer.py

```diff
@@ -164,10 +164,20 @@ def get_plot(self_or_cls, obj, renderer=None):
         """
         # Initialize DynamicMaps with first data item
         initialize_dynamic(obj)
+        backend = self_or_cls.backend
 
-        if not isinstance(obj, Plot) and not displayable(obj):
-            obj = collate(obj)
-            initialize_dynamic(obj)
+        if not isinstance(obj, Plot):
+            if not displayable(obj):
+                obj = collate(obj)
+                initialize_dynamic(obj)
+            if any(len(c._pattern_spec) > 1 for c in Compositor.definitions):
+                obj = obj.map(lambda obj: Compositor.collapse_element(obj, mode='data',
+                                                                      backend=backend),
+                              [CompositeOverlay])
+            if any(len(c._pattern_spec) == 1 for c in Compositor.definitions):
+                obj = obj.map(lambda obj: Compositor.collapse_element(obj, mode='data',
+                                                                      backend=backend),
+                              [Element])
```

Comment:
Maybe this stuff should be provided by Compositor itself? (i.e ``CompositeOverlay`` vs ``Element``)

Reply:
I also think this should be done by compositor as you are accessing an underscore attribute, namely ``_pattern_spec``.